jetbrains-core-gui/tst/software/aws/toolkits/jetbrains/ui/s3/S3BrowserTest.kt

```diff
@@ -0,0 +1,232 @@
+                    rightClick(0, FOLDER)
+                    clickMenuItem { it.text.contains(UPLOAD_ACTION) }
+                    fileChooserDialog {
+                        setPath(testDataPath.resolve("testFiles").resolve(JSON_FILE).toString())
+                        clickOk()
+                    }
+
+                    waitAMoment()
+
+                    doubleClick(0, FOLDER)
+
+                    assertNotNull(findPath(FOLDER, JSON_FILE))
+                }
+
+                step("Rename a file") {
+                    rightClick(0, FOLDER, JSON_FILE)
+                    clickMenuItem { it.text.contains(RENAME_ACTION) }
+
+                    dialog(RENAME_ACTION) {
+                        textfield(null).setText(NEW_JSON_FILE_NAME)
+                        button(OK_BUTTON).clickWhenEnabled()
+                    }
+
+                    waitAMoment()
+
+                    assertNotNull(findPath(FOLDER, NEW_JSON_FILE_NAME))
+                }
+
+                step("Delete a file") {
+                    rightClick(0, FOLDER, NEW_JSON_FILE_NAME)
+                    clickMenuItem { it.text.contains(DELETE_ACTION) }
+
+                    findMessageDialog(DELETE_ACTION).click(DELETE_PREFIX)
+
+                    waitAMoment()
+
+                    assertNull(findPath(FOLDER, NEW_JSON_FILE_NAME))
+                }
+
+                step("Open known file-types") {
+                    doubleClick(0, JSON_FILE)
+
+                    waitAMoment()
+
+                    assertNotNull(FileEditorManager.getInstance(project).allEditors.mapNotNull { it.file }.find {
+                        it.name.contains(JSON_FILE) && it.fileType::class.simpleName?.contains("JsonFileType") == true
+                    })
+                }
+            }
+        }
+    }
+
+    @After
+    fun cleanUp() {
+        step("Delete bucket named $bucket") {
+            ideFrame {
+                toolwindow("aws.explorer") {
+                    with(jTree(S3_NAME).expandPath(S3_NAME)) {
+                        path(S3_NAME, bucket).rightClickPath()
+                    }
+
+                    clickMenuItem { it.text.contains(DELETE_PREFIX) }
+                    dialog(DELETE_PREFIX, predicate = Predicate.startWith) {
+                        textfield(null).setText(bucket)
+                        button(OK_BUTTON).clickWhenEnabled()
+                    }
+
+                    waitAMoment()
+                }
+            }
+        }
+    }
+
+    private fun TreeTableFixture.findPath(vararg paths: String) = try {
+        ExtendedJTreePathFinder(target().tree).findMatchingPath(paths.toList())
+    } catch (_: Exception) {
+        null
+    }
+
+    private fun IdeFrameFixture.treeTable(block: TreeTableFixture.() -> Unit) {
+        block(TreeTableFixture(robot(), findComponentWithTimeout(this.target(), TreeTable::class.java)))
+    }
+
+    /** Copied from [com.intellij.testGuiFramework.fixtures.TreeTableFixture] **/
```

Comment:
Why did these need to be copied vs using the TreeTableFixture directly?

Reply:
it only supported single click.

Reply:
Ah, i read it as a straight copy-paste